client/connections/ConnectionToCore.ts

```diff
@@ -181,15 +181,15 @@ export default abstract class ConnectionToCore extends TypedEventEmitter<{
     return this.isDisconnecting === false && this.coreSessions.hasAvailability();
   }
 
-  public async createSession(options: ICreateSessionOptions): Promise<CoreSession> {
+  public async createSession(options: ICreateSessionOptions): Promise<CoreSession | Error> {
     try {
       const sessionMeta = await this.commandQueue.run<ISessionMeta>('Session.create', options);
       const session = new CoreSession({ ...sessionMeta, sessionName: options.sessionName }, this);
       this.coreSessions.track(session);
       return session;
     } catch (error) {
       if (error instanceof DisconnectedFromCoreError && this.isDisconnecting) return null;
-      throw error;
+      return error;
```

Comment:
Is this a new pattern? Returning errors instead of throwing them?

Reply:
It's not new... this is part of the unfortunate reality that if you throw this error and no one is looking, node shits itself.

Reply:
But you're catching the errors down in the Agent class.

Reply:
I was trying to track down an unhandled exception. I can switch this back and re-catch in Agent if you want to minimize locations we do this weird thing.